packages/open-payments/src/client/incoming-payment.test.ts

```diff
@@ -94,6 +110,95 @@ describe('incoming-payment', (): void => {
     })
   })
 
+  describe('createIncomingPayment', (): void => {
+    test.each`
+      incomingAmount                                      | expiresAt                                      | description  | externalRef
+      ${undefined}                                        | ${undefined}                                   | ${undefined} | ${undefined}
+      ${{ assetCode: 'USD', assetScale: 2, value: '10' }} | ${new Date(Date.now() + 60_000).toISOString()} | ${'Invoice'} | ${'#INV-1'}
+    `(
+      'returns the incoming payment on success',
+      async ({
+        incomingAmount,
+        expiresAt,
+        description,
+        externalRef
+      }): Promise<void> => {
+        const incomingPayment = mockIncomingPayment({
+          incomingAmount,
+          expiresAt,
+          description,
+          externalRef
+        })
+
+        nock(baseUrl).post('/incoming-payment').reply(200, incomingPayment)
+
+        const result = await createIncomingPayment(
+          {
+            axiosInstance,
+            logger
+          },
+          {
+            url: `${baseUrl}/incoming-payment`,
+            body: {
+              incomingAmount,
+              expiresAt,
+              description,
+              externalRef
+            }
+          },
+          openApiValidators.successfulValidator
+        )
+
+        expect(result).toEqual(incomingPayment)
+      }
+    )
+
+    test('throws if the created incoming payment does not pass validation', async (): Promise<void> => {
+      const receivedAmount = {
+        assetCode: 'USD',
+        assetScale: 2,
+        value: '10'
+      }
+
+      const incomingPayment = mockIncomingPayment({
+        receivedAmount
+      })
```

Comment:
I had to do some digging to understand why the incoming payment was invalid.
How about something like this to make it explicit?
```suggestion
      const amount = {
        assetCode: 'USD',
        assetScale: 2,
        value: '10'
      }

      const incomingPayment = mockIncomingPayment({
        incomingAmount: amount,
        receivedAmount: amount,
        completed: false
      })
```

Reply:
:thinking: Should we actually do extra validation create incoming payment response?
Disallow non-zero `receivedAmount` / `completed`?

Reply:
Do you mean non-zero `incomingAmount`?

Reply:
non-zero `receivedAmount.value` on a newly created incoming payment

Reply:
Ah I see now, yeah, I think it would be good to add those additional checks :+1: